src/main/java/twilight/AddCommand.java

```diff
@@ -0,0 +1,36 @@
+package twilight;
+
+import java.io.IOException;
+
+/**
+ * Represents a Command to add a particular instance of an event.
+ */
+public class AddCommand extends Command {
+    protected int type;
+    protected String details;
+
+    public AddCommand(int type, String details) {
+        this.type = type;
+        this.details = details;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Storage storage) throws InvalidInputException {
+        String addition = "";
+        if (type == 3) {
+            addition = tasks.add(new Todo(details));
+        } else if (type == 4) {
+            String[] split = details.split(" /from | /to ");
+            addition = tasks.add(new Event(split[0], split[1], split[2]));
+        } else {
+            String[] split = details.split(" /by ");
+            addition = tasks.add(new Deadline(split[0], split[1]));
+        }
+        System.out.println(addition);
```

Comment:
Since you already created the `Ui` abstraction with a `print`, should you use that instead? Or should you abstract out all the UI handling into the `ui` object instantitated in the Twilight` class?

Reply:
Yeah I think I should. I just didn't do it thus far out of laziness.